lib/WorseReflection/Core/Reflection/Collection/ClassLikeReflectionMemberCollection.php

```diff
@@ -0,0 +1,338 @@
+                foreach ($member->propertyElements as $propertyElement) {
+                    foreach ($propertyElement as $variable) {
+                        if ($variable instanceof AssignmentExpression) {
+                            $variable = $variable->leftOperand;
+                        }
+
+                        if (false === $variable instanceof Variable) {
+                            continue;
+                        }
+                        $property = new ReflectionProperty($serviceLocator, $classLike, $member, $variable);
+                        $new->properties[$property->name()] = $property;
+                        $new->items[$property->name()] = $property;
+                    }
+                }
+                continue;
+            }
+            if ($member instanceof MethodDeclaration) {
+                $method = new ReflectionMethod($serviceLocator, $classLike, $member);
+                $new->items[$method->name()] = $method;
+                $new->methods[$method->name()] = $method;
+
+                // promoted properties
+                if ($method->name() === '__construct') {
+                    $parameters = $member->parameters;
+                    /** @phpstan-ignore-next-line */
+                    if (!$parameters) {
+                        continue;
+                    }
+                    $children = $parameters->children;
+                    if (!$children) {
+                        continue;
+                    }
+                    foreach (array_filter($children, function ($member) {
+                        if (!$member instanceof Parameter) {
+                            return false;
+                        }
+                        return $member->visibilityToken !== null;
+                    }) as $promotedParameter) {
+                        if (!$promotedParameter instanceof Parameter) {
+                            continue;
+                        }
+                        $property = new ReflectionPromotedProperty($serviceLocator, $classLike, $promotedParameter);
+                        $new->items[$property->name()] = $property;
+                        $new->properties[$property->name()] = $property;
+                    }
+                }
+                continue;
+            }
+
+            if ($member instanceof EnumCaseDeclaration && $classLike instanceof ReflectionEnum) {
+                $enumCase = new PhpactorReflectionEnumCase($serviceLocator, $classLike, $member);
+                $new->items[$enumCase->name()] = $enumCase;
+                $new->enumCases[$enumCase->name()] = $enumCase;
+                continue;
+            }
+        }
+
+
+        return $new;
+    }
+
+    private function filter(Closure $closure): self
+    {
+        $new = new self([]);
+        foreach (self::MEMBER_TYPES as $collection) {
+            $new->$collection = array_filter($this->$collection, $closure);
```

Comment:
thus and `map()` get called _alot_. wonder if it's faster to use a foreach loop to filter / map?